manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml

```diff
@@ -2882,6 +2882,150 @@ spec:
 ---
+# Source: istiod/templates/telemetryv2_1.18.yaml
+# Note: http stats filter is wasm enabled only in sidecars.
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+  name: stats-filter-1.18
+  namespace: istio-system
+  labels:
+    istio.io/rev: default
+spec:
+  priority: -1
+  configPatches:
+    - applyTo: HTTP_FILTER
+      match:
+        context: SIDECAR_OUTBOUND
+        proxy:
+          proxyVersion: '^1\.17.*'
+        listener:
+          filterChain:
+            filter:
+              name: "envoy.filters.network.http_connection_manager"
+              subFilter:
+                name: "envoy.filters.http.router"
+      patch:
+        operation: INSERT_BEFORE
+        value:
+          name: istio.stats
+          typed_config:
+            "@type": type.googleapis.com/udpa.type.v1.TypedStruct
+            type_url: type.googleapis.com/stats.PluginConfig
+            value:
+              {}
+    - applyTo: HTTP_FILTER
+      match:
+        context: SIDECAR_INBOUND
+        proxy:
+          proxyVersion: '^1\.17.*'
+        listener:
+          filterChain:
+            filter:
+              name: "envoy.filters.network.http_connection_manager"
+              subFilter:
+                name: "envoy.filters.http.router"
+      patch:
+        operation: INSERT_BEFORE
+        value:
+          name: istio.stats
+          typed_config:
+            "@type": type.googleapis.com/udpa.type.v1.TypedStruct
+            type_url: type.googleapis.com/stats.PluginConfig
+            value:
+              {
+                "disable_host_header_fallback": true
+              }
+    - applyTo: HTTP_FILTER
+      match:
+        context: GATEWAY
+        proxy:
+          proxyVersion: '^1\.17.*'
+        listener:
+          filterChain:
+            filter:
+              name: "envoy.filters.network.http_connection_manager"
+              subFilter:
+                name: "envoy.filters.http.router"
+      patch:
+        operation: INSERT_BEFORE
+        value:
+          name: istio.stats
+          typed_config:
+            "@type": type.googleapis.com/udpa.type.v1.TypedStruct
+            type_url: type.googleapis.com/stats.PluginConfig
+            value:
+              {
+                "disable_host_header_fallback": true
+              }
+---
+# Source: istiod/templates/telemetryv2_1.18.yaml
+# Note: tcp stats filter is wasm enabled only in sidecars.
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+  name: tcp-stats-filter-1.18
+  namespace: istio-system
+  labels:
+    istio.io/rev: default
+spec:
+  priority: -1
+  configPatches:
+    - applyTo: NETWORK_FILTER
+      match:
+        context: SIDECAR_INBOUND
+        proxy:
+          proxyVersion: '^1\.17.*'
+        listener:
+          filterChain:
+            filter:
+              name: "envoy.filters.network.tcp_proxy"
+      patch:
+        operation: INSERT_BEFORE
+        value:
+          name: istio.stats
```

Comment:
@zirain @kyessenov I remember we generate this inline now, do we still need this envoyfilter?

Reply:
we didn't finish migrating everything over

Reply:
Another question: should remove older version envoyfilter like  1.13  1.14?

Reply:
I know in the past we kept some number, but I'm not sure how many. I noticed 1.16 did not remove any and just added. I can look at 1.15.

Reply:
So, it looks like when we moved to 1.15 we removed those before 1.13. I assume this is because we support upgrading from 1.3 to 1.15? For 1.16, we simply added and didn't remove. 

So I think we can either just add or I can also remove 1.13 and 1.14 keeping 1.15 1.16 and 1.17. Thoughts?

Reply:
Agree

Reply:
commit added removing 1.13, 1.4, and 1.15 keeping 3, 1.16, 1.17, and 1.18, like we had for 1.15 (we didn't remove going to 1.16 or 1.17, so going to 1.18 we should remove 3 (1 for this plus 2 for the missed ones).